cdap-docs/developer-manual/source/building-blocks/schedules.rst

```diff
@@ -8,169 +8,246 @@
+- Replacing ``abortIfNotMet()`` with ``waitUntilMet()`` would have the effect that the
+  workflow execution is delayed until no other concurrent run of the same workflow is
+  executing.
+- This schedule does not specify properties for the workflow execution.
+- This schedule does not configure a timeout, such that the default timeout of one day
+  is used.
+
+Schedules can be added and controlled by the :ref:`CDAP CLI <cli>` and the :ref:`Lifecycle
+HTTP RESTful API <http-restful-api-lifecycle>`. The :ref:`status of a schedule
+<http-restful-api-lifecycle-status>` can be retrieved, and individual schedules can be
+:ref:`added <http-restful-api-lifecycle-schedule-add>`,
+:ref:`enabled, or disabled <http-restful-api-lifecycle-schedule-disable-enable>`.
+
+When a schedule is initially deployed, it is in a disabled state; it needs to be enabled before
+it starts executing workflows.
+
+.. _schedules-jobs-lifecycle:
+
+Jobs and their Lifecycle
+========================
+
+Execution of workflows is initiated when the trigger of a schedule fires. This creates
+a *Job* for the workflow in the scheduler's *Job Queue*. This job will not necessarily
+execute immediately; instead, each job goes through a life cycle:
+
+- When a job is initially created, it is in state *pending trigger*. Most triggers are
+  fulfilled immediately when the job is created. But some triggers may require additional
+  input: For example, a partition trigger can specify a minimum number of new partitions
+  to be present in a dataset. When one or more partitions are added to the dataset, then
+  this creates an event that leads to the creation of a job. But the number of partitions
+  may not be sufficient yet, and it would require more partition events until the trigger
+  is fulfilled. Until then, the job will remain in *pending trigger* state.
+- When the job’s trigger is fulfilled, the job’s state changes to *pending constraints*.
+  If the job has no constraints, then it will not remain in this state; however, if it
+  has constraints, then it remains *pending constraints* until all constraints are fulfilled.
+  The scheduling system now continuously checks whether its constraints are fulfilled.
+  During this check, if any constraint is not fulfilled and it was added with
+  ``abortIfNotMet()``, then the job is canceled and removed from the job queue.
+- When all of a job’s constraints are fulfilled, the job’s state changes to *pending launch*.
+  At this time, the scheduling system will prepare the execution of the workflow, and once
+  it is started successfully, the job is complete and removed from the job queue. Note that
+  the workflow itself can still fail during its execution, but if the scheduler
+  successfully submitted the workflow for execution, then the job is considered complete from
+  the scheduler’s point of view. If starting the workflow fails, however, the job remains
+  *pending launch* and the system will retry execution.
```

Comment:
forever... that might be a future improvement. Created https://issues.cask.co/browse/CDAP-11798.